packages/node/src/soroban/soroban.server.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { getLogger } from '@subql/node-core';
+import { Server, SorobanRpc } from 'soroban-client';
+import { GetEventsRequest } from 'soroban-client/lib/server';
+
+const logger = getLogger('soroban-server');
+const DEFAULT_PAGE_SIZE = 100;
+
+export class SorobanServer extends Server {
```

Comment:
Can we have tests for this please. 

Its a bit hard to know the behaviour if there are more than `DEFAULT_PAGE_SIZE` events ion a block.

Reply:
to reduce the probability of events getting paginated out, we can set the page limit to maximum value (10000). it there were more than 10000 events in a block, the rpc wouldn't return the events pass 10000 index anyway.

Reply:
it would be balanced out by the cache on subsequent requests?